### PR TITLE
Preserve active state on mouseout when the mouse is held

### PR DESCRIPTION
**Pull request by david-davidson**
Fixes https://github.com/FormidableLabs/radium/issues/66

This adds a helper module (`mixins/util/global-mouseup`) that sets up a single window-level `mouseup` listener. Components subscribe to (and unsubscribe from) it behind the scenes via `mixins/browser-state`, which now has a `handleGlobalMouseUp` method. Finally, `_handleMouseLeave` no longer sets the `active` state to false.

/cc @alexlande

### CODE REVIEW

**Comment by alexlande** [comment]
Going to kick the tires on this one, but looks awesome so far.

File: modules/mixins/util/global-mouseup.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+var _mouseUpListenerIsActive = false;
+
+var _handleMouseUp = function (ev) {
+  _listeningComponents.forEach(function (component) {
+    component.handleGlobalMouseUp(ev);
+  });
+};
+
+var subscribeToGlobalMouseUp = function (component) {
+  if (!includes(_listeningComponents, component)) {
```
Comment (ryan-roemer) [inline modules/mixins/util/global-mouseup.js]: I think you can avoid this dependency with normal `ARRAY.indexOf()`...

File: modules/mixins/browser-state.js [hunk 2]
```diff
@@ -1,4 +1,17 @@
+var globalMouseUp = require("./util/global-mouseup");
```
Comment (ianobermiller) [inline modules/mixins/browser-state.js]: How about naming the module MouseUpListener and providing a simple method, subscribe, that takes a callback and returns an object with a `remove()` method. Then, BrowserStateMixin is not tightly coupled, and you don't have the magic of handleGlobalMouseUp being called across files.

File: modules/mixins/browser-state.js [hunk 3]
```diff
@@ -59,6 +71,17 @@ var BrowserStateMixin = {
     });
   },
 
+  radiumGlobalMouseUp: function (ev) {
+    if (ev.toElement === this.getDOMNode()) {
```
Comment (alexlande) [inline modules/mixins/browser-state.js]: Given the desired behavior, is there any reason we couldn't remove this check *and* remove the mouseUp behavior in `browser-state.js`? https://github.com/FormidableLabs/radium/blob/master/modules/mixins/browser-state.js#L54-L60

Then the global mouseUp could handle both cases.

File: modules/mixins/util/global-mouseup.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+var _listeningComponents = [];
+var _mouseUpListenerIsActive = false;
```
Comment (alexlande) [inline modules/mixins/util/global-mouseup.js]: What do we need `_mouseUpListenerIsActive` for?

Comment (alexlande) [inline modules/mixins/util/global-mouseup.js]: Scratch that, I get it.

**Comment by david-davidson** [comment]
@alexlande @ianobermiller @ryan-roemer Thanks for the good suggestions!

File: modules/mixins/util/mouse-up-listener.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+var _callbacks = [];
+var _mouseUpListenerIsActive = false;
+
+var _includes = function (arr, item) {
+  return arr.indexOf(item) !== -1;
```
Comment (alexlande) [inline modules/mixins/util/mouse-up-listener.js]: Let's drop this function and just do this check on line 15:

```js
if (arr.indexOf(item) === -1) {
  _callbacks.push(callback);
}
```

**Comment by alexlande** [comment]
Awesome, thanks @david-davidson!

**Pull request merged**